### PR TITLE
[39] Acl value in modify_permissions changed to an enum

### PR DESCRIPTION
**Pull request by sam-i-am012**
Addresses #39. All tests are still passing.

### CODE REVIEW

File: src/main/java/org/irods/Util/Acl.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package org.irods.Util;
+
+/**
+ * enum for modify_permission() method to ensure user only inputs valid data
+ * */
+public enum Acl {
```
Comment (korydraughn) [inline src/main/java/org/irods/Util/Acl.java:6]: ACL (access control list) and Permissions serve the same purpose.

Let's remove `Acl` and use the `Permission` enumeration.

Then the modify_permissions operation can simply throw an exception if the user passes an enumeration that isn't supported.

Comment (sam-i-am012) [reply src/main/java/org/irods/Util/Acl.java:6]: Done. I made an issue to add a test to make sure it throws an exception. 
Created #80

**Comment by sam-i-am012** [comment]
All existing review comments have been resolved.

File: src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java [hunk 2]
```diff
@@ -16,9 +18,19 @@ public class ModifyPermissionsOperations {
...
-        this.acl = acl;
+        switch (acl) {
+            case NULL:
+            case READ:
+            case WRITE:
+            case OWN:
+                this.acl = acl.getValue();
+                break;
+            default:
+                throw new IrodsException("'" + acl.getValue() +
```
Comment (korydraughn) [inline src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java:31]: Let's use `IllegalArgumentException` to help the dev understand what went wrong.

File: src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java [hunk 3]
```diff
@@ -16,9 +18,19 @@ public class ModifyPermissionsOperations {
...
+                        "' is not a supported Permission in modifyPermission operation.");
```
Comment (korydraughn) [inline src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java:32]: Let's tell the dev what the value was expected to be. For example:
```
Expected one of the following Permission enum values: OWN, WRITE, READ, NULL
```

File: src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java [hunk 4]
```diff
@@ -16,9 +18,19 @@ public class ModifyPermissionsOperations {
     @JsonProperty("acl")
     private String acl;
 
-    public ModifyPermissionsOperations(String entity_name, String acl) {
+    public ModifyPermissionsOperations(String entity_name, Permission acl) throws IrodsException {
```
Comment (korydraughn) [inline src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java:21]: Seeing `Permission acl` is a little weird IMO. Please change `acl` to `perm`.

File: src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java (hunk 2 above)
Comment (sam-i-am012) [reply src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java:31]: Done.

File: src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java (hunk 3 above)
Comment (sam-i-am012) [reply src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java:32]: Done.

**Comment by korydraughn** [comment]
There's a rebase/merge conflict with upstream. Please resolve.

File: src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java [hunk 5]
```diff
@@ -18,18 +18,18 @@ public class ModifyPermissionsOperations {
     @JsonProperty("acl")
     private String acl;
 
-    public ModifyPermissionsOperations(String entity_name, Permission acl) throws IrodsException {
+    public ModifyPermissionsOperations(String entity_name, Permission perm) {
```
Comment (korydraughn) [inline src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java:21]: Need to include the `throws` specifier so the dev knows the constructor can throw exceptions.

Comment (sam-i-am012) [reply src/main/java/org/irods/Serialize/ModifyPermissionsOperations.java]: Done.

**Comment by sam-i-am012** [comment]
Done.

**Review by korydraughn — COMMENTED** [review]
If the tests pass, squash it. No pounds.

**Comment by sam-i-am012** [comment]
It's been squashed.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by sam-i-am012** [comment]
Done.

**Comment by korydraughn** [comment]
Merging.

**Pull request merged**